ergotree-interpreter/src/eval/sglobal.rs

```diff
@@ -53,18 +54,115 @@ pub(crate) static XOR_EVAL_FN: EvalFn = |_mc, _env, _ctx, obj, args| {
     }
 };
 
+pub(crate) static SGLOBAL_FROM_BIGENDIAN_BYTES_EVAL_FN: EvalFn = |mc, _env, _ctx, obj, args| {
+    if obj != Value::Global {
+        return Err(EvalError::UnexpectedValue(format!(
+            "sglobal.fromBigEndianBytes expected obj to be Value::Global, got {:?}",
+            obj
+        )));
+    }
+
+    let bytes_val = args
+        .first()
+        .cloned()
+        .ok_or_else(|| EvalError::NotFound("fromBigEndianBytes: missing bytes arg".to_string()))?;
+    let type_val = mc.tpe().t_range.clone();
+
+    let bytes = match bytes_val {
+        Value::Coll(CollKind::NativeColl(NativeColl::CollByte(bytes))) => bytes,
+        _ => {
+            return Err(EvalError::UnexpectedValue(format!(
+                "fromBigEndianBytes: expected first argument to be byte array, got {:?}",
+                bytes_val
+            )))
+        }
+    };
+
+    match *type_val {
+        SType::SByte => {
+            if bytes.len() != 1 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Byte with fromBigEndianBytes, exactly one byte should be provided".to_string(),
+                ));
+            }
+            Ok(Value::Byte(bytes[0]))
+        }
+        SType::SShort => {
+            if bytes.len() != 2 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Short with fromBigEndianBytes, exactly two bytes should be provided".to_string(),
+                ));
+            }
+            let b0 = bytes[0] as i16;
+            let b1 = bytes[1] as i16;
+            Ok(Value::Short(((b0 & 0xFF) << 8 | (b1 & 0xFF)) as i16))
+        }
+        SType::SInt => {
+            if bytes.len() != 4 {
+                return Err(EvalError::UnexpectedValue(
+                    "To deserialize Int with fromBigEndianBytes, exactly four bytes should be provided".to_string(),
+                ));
+            }
+            let bytes_vec: Vec<u8> = bytes.iter().map(|&x| x as u8).collect();
```

Comment:
Better to do this without heap allocating, rust doesn't have any great ways to collect into fixed-size arrays, so maybe a fold that accumulates bytes would be better: something like bytes.iter().map(|b| b as u8).fold(0i32, |acc, x| (acc << 8) || x as i32)

Reply:
Fixed to use fold for heap-free byte accumulation.